processing/src/main/java/org/apache/carbondata/lcm/status/SegmentStatusManager.java

```diff
@@ -410,6 +410,14 @@ public void writeLoadDetailsIntoFile(String dataLoadLocation,
       for (LoadMetadataDetails loadMetadata : listOfLoadFolderDetailsArray) {
 
         if (loadId.equalsIgnoreCase(loadMetadata.getLoadName())) {
+          // if the segment is compacted then no need to delete that.
+          if (CarbonCommonConstants.SEGMENT_COMPACTED
+              .equalsIgnoreCase(loadMetadata.getLoadStatus())) {
+            LOG.error("Cannot delete the load which is compacted.");
```

Comment:
Here logs will be only in case of user is trying to delete the compacted loads intentionally using delete segment DDL.  so it is ok to add the segment ID in logs.
Fixing that.